### PR TITLE
dnos6_config (New Module)

### PR DESCRIPTION
**Pull request by skg-net**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME
- modules/core/network/dnos6/dnos6_config
##### ANSIBLE VERSION
- ansible 2.2.0 (devel a1947401ba) last updated 2016/09/08 17:44:28 (GMT -700) lib/ansible/modules/core: (detached HEAD db38f0c) last updated 2016/09/08 17:44:58 (GMT -700) lib/ansible/modules/extras: (detached HEAD 8bfdcfcab2) last updated 2016/09/08 17:44:58 (GMT -700) config file = /home/skg/lothlorien/ansible.cfg configured module search path = Default w/o overrides
##### SUMMARY
- Added support for dnos9_config module to support Dell Networking OS6 devices.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @skg-net for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by skg-net** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @skg-net for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: network/dnos6/dnos6_config.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    a deterministic way.
+extends_documentation_fragment: dnos6
+options:
+  lines:
+    description:
+      - The ordered set of commands that should be configured in the
+        section.  The commands must be the exact same commands as found
+        in the device running-config.  Be sure to note the configuration
+        command syntax as some commands are automatically modified by the
+        device config parser. This argument is mutually exclusive with O(src).
```
Comment (gundalow) [inline network/dnos6/dnos6_config.py]: There isn't a `O(...)`

`I(src)` for italics
or `C(...)` for code

File: network/dnos6/dnos6_config.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        level or global commands.
+    required: false
+    default: null
+  src:
+    description:
+      - Specifies the source path to the file that contains the configuration
+        or configuration template to load.  The path to the source file can
+        either be the full path on the Ansible control host or a relative
+        path from the playbook or role root dir.  This argument is mutually
+        exclusive with O(lines).
```
Comment (gundalow) [inline network/dnos6/dnos6_config.py]: I(...)

File: network/dnos6/dnos6_config.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    description:
+      - The C(save) argument instructs the module to save the running-
+        config to the startup-config at the conclusion of the module
+        running.  If check mode is specified, this argument is ignored.
+    required: false
+    default: no
+    choices: ['yes', 'no']
+  config:
+    description:
+      - The C(config) argument allows the playbook desginer to supply
```
Comment (gundalow) [inline network/dnos6/dnos6_config.py]: designer

File: network/dnos6/dnos6_config.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        a change needs to be made.  This allows the playbook designer
+        the opportunity to perform configuration commands prior to pushing
+        any changes without affecting how the set of commands are matched
+        against the system.
+    required: false
+    default: null
+  after:
+    description:
+      - The ordered set of commands to append to the end of the command
+        stack if a changed needs to be made.  Just like with I(before) this
```
Comment (gundalow) [inline network/dnos6/dnos6_config.py]: s/changed/change/

**Comment by gundalow** [comment]
needs_revision

File: network/dnos6/dnos6_config.py (hunk 1 above)
Comment (skg-net) [reply network/dnos6/dnos6_config.py]: Done.

File: network/dnos6/dnos6_config.py (hunk 4 above)
Comment (skg-net) [reply network/dnos6/dnos6_config.py]: Done.

File: network/dnos6/dnos6_config.py (hunk 1 above)
Comment (gundalow) [reply network/dnos6/dnos6_config.py]: I don't see a commit with the `I(...)` --> `C(...)` changes in

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
shipit

**Review by gundalow — APPROVED** [review]

**Comment by gregdek** [comment]
Thanks again to @skg-net for this PR, and thanks @ansible/core for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by skg-net** [comment]
Based the infra required for template module. We need to rework on this config module as well.
So I am closing this PR. I will create a new one once, code is ready.

**Comment by gundalow** [comment]
Merged, thanks

**Pull request merged**